src/app/data-planes/views/DataPlaneDetailView.vue

```diff
@@ -234,8 +317,18 @@ const props = defineProps<{
 }>()
 
 const statusWithReason = computed(() => getStatusAndReason(props.data.dataplane, props.data.dataplaneInsight))
-const dataPlaneTags = computed(() => dpTags(props.data.dataplane))
-const dataPlaneVersions = computed(() => getVersions(props.data.dataplaneInsight))
+
+const lastUpdatedTime = computed(() => {
+  const subscriptions = props.data.dataplaneInsight?.subscriptions ?? []
+
+  if (subscriptions.length === 0) {
+    return null
+  }
+
+  const lastSubscription = subscriptions[subscriptions.length - 1]
+
+  return formatIsoDate(lastSubscription.status.lastUpdateTime)
+})
```

Comment:
I can, but I think this belongs to dataplanes. Zones and DPPs use different subscription formats (KDS vs Discovery subscriptions). This field happens to be in the same location structurally, but that isn’t the case for all subscription properties between those two types.

Reply:
Oh I see, it would be great if the same function could be used for both looks like it only needs to be something that is has 'subscriptions[].status.lastUpdateTime' passed into it.

For the moment moving it to `app/data-planes/data/index.ts` would be fine ny me, at least then we are reusing it for the places for dataplanes that we use it.

I still have the "refactor the `/zones/data/index.ts`" file on my list from when I moved stuff over there so it could potentially be part of that if you don't want to move to `subscriptions/data` in this PR.

Reply:
I added this in https://github.com/kumahq/kuma-gui/pull/1745/commits/a3583cb51649429cdd1541923264f4397f026ecd now.

I’m not entirely sure what eventually will be most convenient to accept. Part of me wants to just pass the `DataplaneOverview` in there, but I want to get to a point where we avoid having lots of `props.dataplaneOverview.dataplaneInsight?.subscriptions ?? []` style code inside such functions and so I think this should probably be provided by the call site. This might change again, but for now it doesn’t matter much.

Reply:
Well looking at this now this is a bit clearer with the simpler code how little this is doing. So, on second thoughts this really this could probably just be done inline:

```vue
{{ formatIsoDate(props.dataplaneOverview.dataplaneInsight.subscriptions) }}
```

Of course ideally it would require a centralized data layer with all our fixups to avoid all the `?` but it could be done with out that.

No matter, I think putting this over into the `data/` folder means it will be dealt with when we go through those as a whole.

Just to be clear, no action here, and it is probably going to change in the future anyway.

Feel free to resolve.